source/DoubleCache/Redis/RedisStaleCache.cs

```diff
@@ -119,21 +119,21 @@ public object Get(string key, Type type, Func<object> dataRetriever, TimeSpan? t
             if (timeToLive.HasValue)
                 staleTtl = timeToLive.Value.Add(_staleDuration);
 
-            var item = await _redisCache.GetAsync(key, dataRetriever, staleTtl);
-            var ttl = await _database.KeyTimeToLiveAsync(key);
+            var item = await _redisCache.GetAsync(key, dataRetriever, staleTtl).ConfigureAwait(false);
+            var ttl = await _database.KeyTimeToLiveAsync(key).ConfigureAwait(false);
             if (!ttl.HasValue || ttl.Value < _staleDuration)
             {
                 ttl = ttl == null
                     ? _staleDuration.Add(_staleDuration)
                     : ttl.Value.Add(_staleDuration);
 
-                await _database.KeyExpireAsync(key, ttl);
+                await _database.KeyExpireAsync(key, ttl).ConfigureAwait(false);
 
                 ThreadPool.QueueUserWorkItem(async o =>
                 {
                     try
                     {
-                        _redisCache.Add(key, await dataRetriever.Invoke(), staleTtl);
+                        _redisCache.Add(key, await dataRetriever.Invoke().ConfigureAwait(false), staleTtl);
```

Comment:
I don't think this will play well with `PublishingCache`. This will return the old value, so `executed` will be false, unless somehow `dataRetriever.Invoke()` returns before this method does, which is very unlikely.

I think the way you'd fix this is in the `dataRetriever` in `PublishingCache` you remove `executed` from the method and instead use `Task.ContinueWith(_cachePublisher.NotifyUpdate(...));` to asynchronously notify the other instances.